File: demo-quarkus-spring-api/src/main/java/org/harryng/demo/quarkus/counter/service/CounterService.java
package org.harryng.demo.quarkus.counter.service;

import org.harryng.demo.quarkus.counter.entity.CounterImpl;

public interface CounterService{
    CounterImpl insert(String id, long initValue) throws RuntimeException, Exception;

    long increment(String id, int step) throws RuntimeException, Exception;

    long currentValue(String id) throws RuntimeException, Exception;
}

File: demo-quarkus-spring-api/src/main/java/org/harryng/demo/quarkus/counter/service/CounterService.kt
package org.harryng.communication.counter.service

import org.harryng.communication.counter.entity.CounterImpl
import org.harryng.demo.quarkus.kernel.counter.CounterPersistence
import kotlin.jvm.Throws

interface CounterService{
    @Throws(RuntimeException::class)
    fun insert(id: String, initValue: Long = CounterPersistence.DEFAULT_INIT_VALUE) : CounterImpl

    @Throws(RuntimeException::class)
    fun increment(id:String, step: Int = CounterPersistence.DEFAULT_STEP) : Long

    @Throws(RuntimeException::class)
    fun currentValue(id:String): Long
}